crates/database/src/orders.rs

```diff
@@ -531,6 +531,72 @@ WHERE
     sqlx::query_as(QUERY).bind(tx_hash).fetch(ex)
 }
 
+#[derive(Debug, sqlx::FromRow)]
+pub struct OrderExecution {
+    pub executed_solver_fee: Option<BigDecimal>,
+    pub sell_token: Address,
+    pub buy_token: Address,
+    pub kind: OrderKind,
+    /// The amount of `sell_token`s that actually got converted to `buy_token`
+    /// for the user to receive.
+    pub sell_amount_before_fees: BigDecimal,
+    pub buy_amount: BigDecimal,
+    pub executed_amount: BigDecimal,
+    pub signature: Vec<u8>,
+    pub signing_scheme: SigningScheme,
+    pub owner: Address,
+}
+
+pub fn order_executions_in_tx<'a>(
+    ex: &'a mut PgConnection,
+    tx_hash: &'a TransactionHash,
+) -> BoxStream<'a, Result<OrderExecution, sqlx::Error>> {
+    const QUERY: &str = const_format::formatcp!(
+        r#"
+-- This will fail if we ever have multiple settlements in the same transaction because this WITH
+-- query will return multiple rows. If we ever want to do this, a tx hash is no longer enough to
+-- uniquely identify a settlement so the "orders for tx hash" route needs to change in some way
+-- like taking block number and log index directly.
+WITH settlement AS (
+    SELECT block_number, log_index
```

Comment:
This query shares a lot of structure with the one it was created from. The `with` block and the `where` block are identical. Might be worth factoring that out with the const string concat we're using for queries in some places. Not sure.

Reply:
Yes. I copied those parts one to one. Will try to refactor it and check what it looks like. 👌

Reply:
I implemented this.